hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskValue.java

```diff
@@ -96,12 +95,19 @@ public VirtualValue asReadOnly() {
     /** {@inheritDoc} */
     @Override
     public void serialize(@NonNull final ByteBuffer byteBuffer) throws IOException {
+        serializeReturningWrittenBytes(byteBuffer);
+    }
+
+    public int serializeReturningWrittenBytes(@NonNull ByteBuffer byteBuffer) throws IOException {
+        int initPos = byteBuffer.position();
         final var output = BufferedData.wrap(byteBuffer);
```

Comment:
Today, `output.position()` matches `byteBuffer.position()`, but this is not immediately obvious from reading this code. So my proposal is to wrap the buffer first, and then use its position as `initPos`. The original byte buffer will be used in a single place, to create a wrapping buffered data object, all remaining code will work with `output`.

Reply:
Agreed. Fixed